hugo/content/docs/tutorials/wine.md

```diff
@@ -0,0 +1,256 @@
+$ dud stage gen -o data/
+working-dir: .
+outputs:
+  data:
+    is-dir: true
+```
+
+We also want to tell `dud` _how_ the artifact is generated. We can use `--` to pass our command to the stage.
+```
+$ dud stage gen -o data/ -- wget -q -r -np -nd -A csv https://archive.ics.uci.edu/ml/machine-learning-databases/wine-quality/ -P data
+command: wget -q -r -np -nd -A csv https://archive.ics.uci.edu/ml/machine-learning-databases/wine-quality/
+  -P data
+working-dir: .
+outputs:
+  data:
+    is-dir: true
+```
+
+It looks like this has all of the information we need! However, we haven't actually saved the output stage to a file yet. We can use `tee` to redirect `dud`'s autogenerated stage to a YAML file.
+```
+$ dud stage gen -o data/ -- wget -q -r -np -nd -A csv https://archive.ics.uci.edu/ml/machine-learning-databases/wine-quality/ -P data | tee get_data.yaml
+```
+
+Let's tell `dud` to track that stage and check the status.
+```
+$ dud stage add get_data.yaml
+$ dud status
+get_data.yaml   (stage definition not checksummed)
+  data  incorrect file type: Absent
+```
+
+Notice that the stage hasn't been checksummed, and that `dud` complains that the `data` artifact is absent. Let's fix that by running the stage and then checking the status again.
+```
+$ dud run get_data.yaml
+$ dud st
+get_data.yaml   (stage definition not checksummed)
+  data  uncommitted
+```
+
+We've made some progress! `dud` is aware of the data artifact and knows that is uncommitted. In fact, it a design decision for `dud` to wait to commit until you tell it. This is because the commit operation is one of the mostly costly operations since it involves lots of hashing.
+
+We can tell `dud` that we are in-fact ready to commit and then check the status one last time.
+```
+$ dud commit get_data.yaml
+committing stage get_data.yaml
+⠋ committing data (340 kB, 390.873 MB/s)
+$ dud st
+get_data.yaml   (stage definition up-to-date)
+  data  up-to-date
+```
+
+Excellent! Remember how the cache was empty before? You'll find that after the commit, the cache is no longer empty. 
+```
+$ tree .dud
+.dud
+├── cache
+│   ├── 3f
+│   │   └── 89718d7db7e8983db992bbe63b63c912b510aa279eef55fd6927f98a4a72f5
+│   ├── 9d
+│   │   └── 3fee280415b109adc112deb3d64ef76a9184387737aa841fda4d7687d65c13
+│   └── dc
+│       └── 42702fce68232255dcd5d83d2f5d82395bfdc9ea81c1fc7bf9d9b6e4ebe57e
+├── config.yaml
+├── index
+└── rclone.conf
+
+4 directories, 6 files
+```
+
+Those content-addressed files in the cache correspond to the data in the `data` folder. By default, `dud` symlinks the files in the working directory to point the files in the cache.
+
+What if, through some freak accident, we delete our `data` folder? Since we committed, `dud` has us covered.
+```
+$ rm -r data/
+$ ls
+get_data.yaml
+```
+
+No worries! We can simply recover the folder back to the working directory with a `dud checkout`.
+```
+$ dud checkout get_data.yaml
+checking out stage get_data.yaml
+$ ls
+data  get_data.yaml
+```
+
```

Comment:
I think the end of this section should walk through committing this work to Git. The below can be saved for the next part.